src/core/Akka.Remote/RemoteSystemDaemon.cs

```diff
@@ -185,46 +204,25 @@ protected void OnReceive(object message, IActorRef sender)
                     {
                         _terminating.Locked(() =>
                         {
-                           IImmutableSet<IActorRef> children;
-                            if (_parent2Children.TryRemove(parent,out children))
+                            IImmutableSet<IActorRef> children;
+                            if (_parent2Children.TryRemove(parent, out children))
                             {
                                 foreach (var c in children)
                                 {
                                     _system.Stop(c);
                                     var name = c.Path.Elements.Drop(1).Join("/");
-                                    RemoveChild(name,c);
+                                    RemoveChild(name, c);
                                 }
                                 TerminationHookDoneWhenNoChildren();
                             }
                         });
                     }
-                }               
+                }
             }
-        }
-
-        private void TerminationHookDoneWhenNoChildren()
-        {
-            _terminating.WhileOn(() =>
+            else
             {
-                if (!HasChildren)
-                {
-                    _terminator.Tell(TerminationHookDone.Instance, this);
-                }
-            });
-        }
-
-  //      def terminationHookDoneWhenNoChildren(): Unit = terminating.whileOn {
-  //  if (!hasChildren) terminator.tell(TerminationHookDone, this)
-  //}
-
-        /// <summary>
-        ///     Tells the internal.
-        /// </summary>
-        /// <param name="message">The message.</param>
-        /// <param name="sender">The sender.</param>
-        protected override void TellInternal(object message, IActorRef sender)
-        {
-            OnReceive(message, sender);
+                base.SendSystemMessage(message);
```

Comment:
Fixed the `RemoteSystemDaemon` to use the `SendSystemMessage` capabilities of the `MinimalActorRef` correctly.